agent/eni/watcher_linux.go

```diff
@@ -131,8 +124,48 @@ func (udevWatcher *UdevWatcher) reconcileOnce() {
 	// As we postulate the netlinkClient.LinkList() call to be expensive, we allow
 	// the race here. The state would be corrected during the next reconciliation loop.
 
-	udevWatcher.state.Reconcile(currentState)
-	log.Debug("Udev watcher reconciliation: end")
+	// Add new interfaces next
+	for mac, _ := range currentState {
+		udevWatcher.sendENIStateChange(mac)
```

Comment:
I specifically want the blocking behavior here as any issues with the channel receiver will result in a deadlock rather than a go routine leak.